src/test/scala/com/microsoft/hyperspace/index/E2EHyperspaceRulesTest.scala

```diff
@@ -224,11 +223,11 @@ class E2EHyperspaceRulesTest extends QueryTest with HyperspaceSuite {
   }
 
   test("E2E test for join query with alias columns is not supported.") {
-    def verifyNoChange(f: () => DataFrame): Unit = {
-      spark.disableHyperspace()
-      val originalPlan = f().queryExecution.optimizedPlan
-      val updatedPlan = JoinIndexRule(originalPlan)
-      assert(originalPlan.equals(updatedPlan))
+    def verifyNoShuffleRemovalFromJoinIndex(f: () => DataFrame): Unit = {
+      spark.enableHyperspace()
+      val updatedPlan = f().queryExecution.executedPlan
+      // indexes will be applied because of FilterIndex
+      assert(updatedPlan.collect{case s: ShuffleExchangeExec => true}.size === 2)
     }
```

Comment:
Is the test description still valid? How do we know that it works as described, or the test just passes because FilterIndex is applied and it prevented JoinIndex from being applied?